src/System.Text.Encoding/tests/Encoder/EncoderConvert2.cs

```diff
@@ -188,6 +188,145 @@ public void PosTest9()
             VerificationHelper(encoder, chars, 0, 1, bytes, 0, bytes.Length, false, 1, 2, true, "009.3");
             VerificationHelper(encoder, chars, 0, 1, bytes, 0, bytes.Length, true, 1, 2, true, "009.4");
         }
+        
+        // PosTest10: Call Convert to convert a arbitrary character array with ASCII encoder
+        [Fact]
+        public void PosTest10()
+        {
+            char[] chars = new char[c_SIZE_OF_ARRAY];
+            byte[] bytes = new byte[c_SIZE_OF_ARRAY];
+            Encoder encoder = Encoding.ASCII.GetEncoder();
+
+            for (int i = 0; i < chars.Length; ++i)
+            {
+                chars[i] = _generator.GetChar(-55);
+            }
+
+            int charsUsed;
+            int bytesUsed;
+            bool completed;
+            encoder.Convert(chars, 0, chars.Length, bytes, 0, bytes.Length, false, out charsUsed, out bytesUsed, out completed);
+
+            // set flush to true and try again
+            encoder.Convert(chars, 0, chars.Length, bytes, 0, bytes.Length, true, out charsUsed, out bytesUsed, out completed);
+        }
+        
+        // PosTest11: Call Convert to convert a ASCII character array with ASCII encoder
+        [Fact]
+        public void PosTest11()
+        {
+            char[] chars = "TestLibrary.TestFramework.BeginScenario".ToCharArray();
+            byte[] bytes = new byte[chars.Length];
+            Encoder encoder = Encoding.ASCII.GetEncoder();
+
+            VerificationHelper(encoder, chars, 0, chars.Length, bytes, 0, bytes.Length, false, chars.Length, chars.Length, true, "011.1");
+            VerificationHelper(encoder, chars, 0, chars.Length, bytes, 0, bytes.Length, true, chars.Length, chars.Length, true, "011.2");
+            VerificationHelper(encoder, chars, 0, 0, bytes, 0, 0, true, 0, 0, true, "011.3");
+        }
+        
+        // PosTest12: Call Convert to convert partial of a ASCII character array with ASCII encoder
+        [Fact]
+        public void PosTest12()
```

Comment:
instead of writing multiple test cases like this, we can write just one theory test cases and supply the data to the method. this will simplify the tests a lot.